demo/site/src/blocks/CookieSafeYouTubeVideoBlock.tsx

```diff
@@ -0,0 +1,25 @@
+import { CookieSafe, PropsWithData, YouTubeVideoBlock } from "@comet/cms-site";
+import { YouTubeVideoBlockData } from "@src/blocks.generated";
+import { CookieFallback } from "@src/components/common/CookieFallback";
+import { cookieIds, useCookieApi } from "@src/util/cookies";
+import styled from "styled-components";
+
+// TODO: Consider if `CookieSafe` should be used directly inside `YouTubeVideoBlock` and expose the `consented` and `fallback` props.
+// - Problem: The `fallback` may sometimes need to be rendered outside of the `YouTubeVideoBlock`.
```

Comment:
For example, a block that has:
- multiple possible videos, only some of which require cookie-consent
- shows a preview that must be rendered outside of the video
- should replace the entire block with the `CookieFallback` if cookies are required but not consented to.

```tsx
<GenericVideoBlock>
    {cookiesAccepted || videoType === "dam" ? (
        <>
            {playingVideo ? (
                <>
                    {videoType === "youtube" && <YouTubeVideoBlock />}
                    {videoType === "dam" && <DamVideoBlock />}
                </>
            ) : (
                <Preview>
                    <PreviewImage />
                    <PlayButton onClick={() => setPlayingVideo(true)} />
                </Preview>
            )}
        </>
    ) : (
        <CookieFallback />
    )}
</GenericVideoBlock>
```

A possible use-case would be the `MediaBlock` in Starter. 
Does this make sense?

One possible workaround would be to always tell the `YouTubeVideoBlock` that cookies have been consented to and instead wrap the `GenericVideoBlock` with another `CookieSafe` component, that then handles this, based on the consented cookies and the `videoType` .

Reply:
Currently all video blocks in the library have their own preview image. I'm not sure if your scenario applies to the media block 🤔

Reply:
I think this is irrelevant for now. 
It should be fine this way, as the workaround I mentioned above should always be possible, in case this occurs. 

We can think of a solution, if it turns out the workaround is actually needed and if it's not a satisfactory solution.